llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -479,6 +486,159 @@ int AIEPostRASchedStrategy::getMaxDeltaCycles(const SchedBoundary &Zone) const {
                    BottomUpDelta.getValue()});
 }
 
+static bool checkSlotConflict(const unsigned OpCodeA, const unsigned OpCodeB,
+                              const AIEBaseMCFormats &Formats) {
+
+  MCSlotKind SlotKindA = Formats.getSlotKind(OpCodeA);
+  MCSlotKind SlotKindB = Formats.getSlotKind(OpCodeB);
+
+  if (SlotKindA != MCSlotKind() && SlotKindB != MCSlotKind()) {
```

Comment:
nit: we can create a "default slot" here, so we don't need to instantiate again.

Reply:
What about:
```
static bool checkSlotConflict(const unsigned OpCodeA, const unsigned OpCodeB,
                              const AIEBaseMCFormats &Formats) {

  const MCSlotKind SlotKindA = Formats.getSlotKind(OpCodeA);
  const MCSlotKind SlotKindB = Formats.getSlotKind(OpCodeB);
  const MCSlotKind UnknownSlot = MCSlotKind();

  if (SlotKindA == UnknownSlot || SlotKindB == UnknownSlot)
    return true;

  return (Formats.getSlotInfo(SlotKindA)->getSlotSet() &
            Formats.getSlotInfo(SlotKindB)->getSlotSet());
}
```